pywbemtools/pywbemlistener/_cmd_listener.py

```diff
@@ -0,0 +1,1159 @@
+# (C) Copyright 2021 Inova Development Inc.
+# All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Click definitions for the pywbemlistener commands.
+
+NOTE: Commands are ordered in help display by their order in this file.
+"""
+
+from __future__ import absolute_import, print_function
+
+import sys
+import os
+import shlex
+import subprocess
+import signal
+import atexit
+import threading
+import argparse
+import importlib
+from time import sleep
+from datetime import datetime
+import click
+import psutil
+import six
+from nocasedict import NocaseDict
+
+from pywbem import WBEMListener, ListenerError
+
+from .._click_extensions import PywbemtoolsCommand, CMD_OPTS_TXT
+from .._options import add_options, help_option, validate_required_arg
+from .._output_formatting import format_table
+
+from . import _config
+from .pywbemlistener import cli
+
+# Signals used for having the 'run' command signal startup completion
+# back to its parent 'start' process.
+# The default handlers for these signals are replaced.
+# The success signal can be any signal that can be handled.
+# In order to handle keyboard interrupts during password prompt correctly,
+# the failure signal must be SIGINT.
+try:
+    # Unix/Linux/macOS
+    # pylint: disable=no-member
+    SIGNAL_RUN_STARTUP_SUCCESS = signal.SIGUSR1
+except AttributeError:
+    # native Windows
+    # pylint: disable=no-member
+    SIGNAL_RUN_STARTUP_SUCCESS = signal.SIGBREAK
+SIGNAL_RUN_STARTUP_FAILURE = signal.SIGINT
+
+# Status and condition used to communicate the startup completion status of the
+# 'run' process between the signal handlers and other functions in the
+# 'start' process.
+RUN_STARTUP_STATUS = None
+RUN_STARTUP_COND = threading.Condition()
+
+# Timeout in seconds for the 'run' command starting up. This timeout
+# also ends a possible prompt for the password of the private key file.
+RUN_STARTUP_TIMEOUT = 60
+
+DEFAULT_LISTENER_PORT = 25989
+DEFAULT_LISTENER_PROTOCOL = 'https'
+DEFAULT_INDI_FORMAT = '{dt} {h} {c} {p}'
+
+LISTEN_OPTIONS = [
+    click.option('--port', type=int, metavar='PORT',
```

Comment:
Can we add single char short option for al least the most used of these options (port, protocol, display)

Reply:
How abobut this change:
* Change `--protocol` to `--scheme` and add `-s`
* Add `-p` for `--port`
* Add `-d` for `--indi-display` but that becomes inconsistent if we want to extend it to call due to `-c` being used for `--certfile` already, so maybe `--id`?

Reply:
DONE.